src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const interArr = sourceString.split(';');
+
+  const arr = interArr
+    .filter((item) => item.trim().length > 0)
+    .map((item) => item.trim().replace(/\s*:\s*/g, ':'));
+
+  const finalObj = arr.reduce((accumulator, item) => {
+    const [key, value] = item.split(':');
+
+    accumulator[key] = value;
+
+    return accumulator;
```

Comment:
[CODE KNOWLEDGE] - Don't use iteration methods to modify an already existing array/object. In this case, the use of `reduce` is appropriate as it is being used to transform an array into an object, and not to modify the array itself.